Nodes/CV7Node.cpp

```diff
@@ -161,52 +174,60 @@ void CV7Node::WindSensorThread(void* nodePtr)
 
 				MessagePtr windData = std::make_unique<WindDataMsg>(node->m_MeanWindDir, node->m_MeanWindSpeed, node->m_MeanWindTemp);
 				node->m_MsgBus.sendMessage(std::move(windData));
-
 			}
 		}
-		else
-		{
-			noDataCount++;
-		}
 	}
 }
 
-bool CV7Node::parseString(const char* buffer, float& windDir, float& windSpeed, float& windTemp) const
+bool CV7Node::parseString(std::string& buffer_to_parse, float& windDir, float& windSpeed, float& windTemp) const
 {
 	const int IIMWV = 0;
 	const int WIXDR = 1;
 	bool updated[] = { false, false };
-	char * writalbeBuff;
-	writalbeBuff = const_cast<char *>(buffer);
-	char* split = strtok(writalbeBuff, "$,");
+	/*"$IIMWV,%03.1f,R,%03.1f,M,*AB
+	   $WIXDR,C,%03.1f,C,,*AB",*/
+	std::regex iimwv_regex ("\\$IIMWV,([^,]{0,6}),.,([^,]{0,6}),.,[^\\$]{0,4}\\n?");
+	std::regex wixdir_regex ("\\$WIXDR,.,([^,]{0,6}),.,.?,[^\\$]{0,4}");
 
-	while (split != NULL)
+  // erase everything before first $ -
+  std::size_t found_begin = buffer_to_parse.find_first_of("$");
+  buffer_to_parse.erase(0,found_begin);
+	//----------------------------------
+
+  if (buffer_to_parse.size()>0)/* if there is still thing in the parser*/
```

Comment:
I think the comment might be more understood if it was "If there is data still to be parsed" However I think this might be a wasted line as any novice engineer could understand that if